autoscheduler/frontend/src/types/Grades.ts

```diff
@@ -23,6 +23,18 @@ export default class Grades {
   Q: number;
   X: number;
   constructor(init: GradesInit) {
+    // None of the properties can be null
+    if (init.gpa == null) { throw Error(`Grades.gpa is invalid: ${init.gpa}`); }
```

Comment:
I'm assuming that you're using double-equals here intentionally, to catch `undefined` as well?

Reply:
uh lol no, I honestly meant to do a triple-equals. I do think we should probably catch undefined as well, as that definitely shouldn't happen